backend/src/test/java/io/swagger/service/TestFilmService.java

```diff
@@ -0,0 +1,191 @@
+        verify(filmRepository, Mockito.times(1)).save(randFilm);
+    }
+
+    @Test
+    public void testWhenAddFilmAlreadyExist_ThenThrowError() {
+        Film randFilm = mockFilmDB.get(new Random().nextInt(movieCount));
+        when(filmRepository.existsById(Mockito.any(UUID.class))).thenReturn(true);
+
+        Exception exception = assertThrows(EntityExistsException.class, () -> filmService.addFilm(randFilm));
+        String expected = "Film already exist";
+        String actual = exception.getMessage();
+
+        assertEquals(expected, actual);
+
+        verify(filmRepository, Mockito.times(1)).existsById(randFilm.getId());
+        verify(filmRepository, Mockito.times(0)).save(randFilm);
+    }
+
+    @Test
+    public void testWhenUpdateFilm_ThenFlow() {
+        Film randFilm = mockFilmDB.get(new Random().nextInt(movieCount));
+        when(filmRepository.existsById(Mockito.any(UUID.class))).thenReturn(true);
+        when(filmRepository.save(Mockito.any(Film.class))).thenReturn(randFilm);
+
+        filmService.updateFilm(randFilm);
+
+        verify(filmRepository, Mockito.times(1)).existsById(randFilm.getId());
+        verify(filmRepository, Mockito.times(1)).save(randFilm);
+    }
+
+    @Test
+    public void testWhenUpdateFilmNotExist_ThenThrowError() {
+        Film randFilm = mockFilmDB.get(new Random().nextInt(movieCount));
+        when(filmRepository.existsById(Mockito.any(UUID.class))).thenReturn(false);
+
+        Exception exception = assertThrows(NoSuchElementException.class, () -> filmService.updateFilm(randFilm));
+        String expected = "Film not found";
+        String actual = exception.getMessage();
+
+        assertEquals(expected, actual);
+
+        verify(filmRepository, Mockito.times(1)).existsById(randFilm.getId());
+        verify(filmRepository, Mockito.times(0)).save(randFilm);
+    }
+
+    @Test
+    public void testWhenDeleteFilm_ThenFlow() {
+        Film randFilm = mockFilmDB.get(new Random().nextInt(movieCount));
+        when(filmRepository.existsById(Mockito.any(UUID.class))).thenReturn(true);
+        doNothing().when(filmRepository).deleteById(randFilm.getId());
+
+        filmService.deleteFilm(randFilm.getId());
+
+        verify(filmRepository, Mockito.times(1)).existsById(randFilm.getId());
+        verify(filmRepository, Mockito.times(1)).deleteById(randFilm.getId());
+    }
+
+    @Test
+    public void testWhenDeleteFilmNonExistant_ThenThrow() {
+        Film randFilm = mockFilmDB.get(new Random().nextInt(movieCount));
+        when(filmRepository.existsById(Mockito.any(UUID.class))).thenReturn(false);
+
+        Exception exception = assertThrows(NoSuchElementException.class, () -> filmService.deleteFilm(randFilm.getId()));
+        String expected = "Film not found";
+        String actual = exception.getMessage();
+
+        assertEquals(expected, actual);
+
+        verify(filmRepository, Mockito.times(1)).existsById(randFilm.getId());
```

Comment:
Warum verify?

Reply:
ist Mockito.verify() ist Library und check ob ne gewissen Funktion mit gewissen Parametern gecheckt wird

Reply:
Ich weiß schon was das macht, aber warum hier? Warum willst du in deinem Test: testWhenDeleteFilmNonExistant_ThenThrow()
überprüfen, ob die existById aufgerufen wurde?
Du willst doch überprüfen ob am Ende eine exception geworfen wurde.
Also, du kannst schon gerne die verify-Methode drin lassen, aber der Test testet nicht das was er sagt.

Reply:
Ich möchte dadurch testen dass diese Funktion ist trotz aufgerufen aber die nächste nicht (deswegen 0 times danach). Ist nur Sanity check quazi. Aber kann auch wenig Sinn macht, wenn ich direkt der betroffene Stelle direkt abchecken